lib/note-content-editor.jsx

```diff
@@ -171,6 +220,29 @@ export default class NoteContentEditor extends React.Component {
 		)
 	}
 
+	handleReturn = () => {
+		// matches lines that start with `- ` or `* `
+		// preceded by 0 or more tab characters
+		const listItemRe = /^\t*[-*]\s/;
```

Comment:
On second thought, I think a space after the character (`-` or `*`) should be required